packages/playwright-core/src/server/dispatchers/localUtilsDispatcher.ts

```diff
@@ -51,14 +55,21 @@ export class LocalUtilsDispatcher extends Dispatcher<{ guid: string }, channels.
     tmpDir: string | undefined,
     callStacks: channels.ClientSideCallMetadata[]
   }>();
+  private _requestContext: APIRequestContext;
 
   constructor(scope: RootDispatcher, playwright: Playwright) {
     const localUtils = new SdkObject(playwright, 'localUtils', 'localUtils');
     const deviceDescriptors = Object.entries(descriptors)
         .map(([name, descriptor]) => ({ name, descriptor }));
+
+    const requestContext = new GlobalAPIRequestContext(
+        playwright,
+        {} // TODO: this should probably respect _combinedContextOptions from test runner
```

Comment:
It should! The way to make it work is to only create context when creating a new mocking proxy, and pass arguments there.